mmdet/models/detectors/base.py

```diff
@@ -213,8 +213,10 @@ def forward(self, img, img_metas, return_loss=True, **kwargs):
         else:
             return self.forward_test(img, img_metas, **kwargs)
 
-    def forward_export(self, imgs):
+    def forward_export(self, imgs, img_metas=None):
         from torch.onnx.operators import shape_as_tensor
+        if img_metas is not None:
+                self.img_metas = img_metas
```

Comment:
```suggestion
    def forward_export(self, imgs, img_metas=None):
        if img_metas is not None and self.img_metas is not None:
            raise RuntimeError('forward_export is called with non-zero img_metas inside forward_export_context')
        from torch.onnx.operators import shape_as_tensor
        if img_metas is None:
                img_metas = self.img_metas
```

And, please, everywhere below replace `self.img_metas` with `img_metas`.
I think that this function should not set or replace the field `self.img_metas` -- IMHO it should be done in special places (I mean, such methods as `forward_export_context` and `forward_nncf_initialization_context`)

Reply:
Done.